app/services/ui_notifications/rss_notifications_checker.rb

```diff
@@ -0,0 +1,50 @@
+require 'rss'
+require 'date'
+
+module UINotifications
+  class RssNotificationsChecker
+    def initialize(options = [])
+      @url = options[:url] || 'https://theforeman.org/feed.xml'
+      @latest_posts = options[:latest_posts] || 3
+      @force_repost = options[:force_repost] || false
+      @audience = options[:audience] || Notification::AUDIENCE_GLOBAL
+    end
+
+    def deliver!
+      feed = RSS::Parser.parse(@url, false)
+      feed.items[0, @latest_posts].each do |item|
+        blueprint = rss_notification_blueprint
+        if notification_already_exists?(item)
+          next unless @force_repost
+        end
+        initiator = User.anonymous_admin
+        notification = Notification.new(
+          :initiator => initiator,
+          :audience => @audience,
+          :message => item.title.content,
+          :notification_blueprint => blueprint,
+          :actions => {
+            :links => [
+              {
+                :href => item.link.href,
+                :title => _('Link to blog'),
+                :external => true
+              }
+            ]
+          }
+        )
+        notification.save
+      end
+    end
+
+    private
+
+    def rss_notification_blueprint
+      NotificationBlueprint.unscoped.find_by_name('rss_post')
+    end
+
+    def notification_already_exists?(item)
+      !!Notification.unscoped.find_by_message(item.summary.content)
```

Comment:
do we know if content is always smaller than message? does it make sense to hash it instead?